project/ScalafixBuild.scala

```diff
@@ -220,6 +220,14 @@ object ScalafixBuild extends AutoPlugin with GhpagesKeys {
         case _ => Seq()
       }
     },
+    Compile / unmanagedResourceDirectories ++= {
+      val resourceDir = (Compile / resourceDirectory).value
+      CrossVersion.partialVersion(scalaVersion.value) match {
+        case Some((2, _)) => Seq(resourceDir / "resource-2")
+        case Some((3, _)) => Seq(resourceDir / "resource-3")
+        case _ => Seq()
+      }
+    },
```

Comment:
```
sbt:example> show Compile / unmanagedResourceDirectories
[info] * /home/piquerez/github/scalacenter/sbt-resource-dir/src/main/resources
[info] * /home/piquerez/github/scalacenter/sbt-resource-dir/src/main/resources/resource-3
```
I don't think it is a good idea to have `resource-3` inside `resources` because they are both `unmanagedResourceDirectories`. Maybe you should put `resource-3` on the same level as `resources`.

```diff
Compile / unmanagedResourceDirectories ++= {
      val resourceDir = (Compile / resourceDirectory).value
      CrossVersion.partialVersion(scalaVersion.value) match {
        case Some((2, _)) => Seq(resourceDir.getParentFile / "resource-2")
        case Some((3, _)) => Seq(resourceDir.getParentFile / "resource-3")
        case _ => Seq()
      }
    },
```

Reply:
Thanks! I'm new to sbt and didn't realize I could actually check the dirs with `show Compile / unmanagedResourceDirectories`. 
On the multiple resource directories or not I'm not sure, we might have resources that are common and only some specific for each scala version. Right now it's not the case so this might be overcomplicating for [speculative generality](https://refactoring.guru/smells/speculative-generality)

What do you think @bjaglin ?

Reply:
If I see a `resource-x` directory in a project, I would probably expect `resource` to be included as well, just like the built-in `scala-x`  pattern for `unmanagedSourceDirectories`. So based on my own experience with sbt projects, I would favor adding new directories rather than replacing the default one. Also, it means we can apply the setting across all modules (even those that don't have `resource-x` needs) instead of on a per-module basis.